nixos/tests/systemd-analyze.nix

```diff
@@ -0,0 +1,41 @@
+import ./make-test-python.nix ({ pkgs, latestKernel ? false, ... }:
+
+{
+  name = "systemd-analyze";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ raskin ];
+  };
+
+  machine =
+    { pkgs, lib, ... }:
+    { boot.kernelPackages = lib.mkIf latestKernel pkgs.linuxPackages_latest;
+      sound.enable = true; # needed for the factl test, /dev/snd/* exists without them but udev doesn't care then
+    };
+
+  testScript = ''
+    machine.wait_for_unit("multi-user.target")
+
+    with subtest("Prepare output dir"):
+        machine.succeed("mkdir systemd-analyze")
+
+
+    def run_systemd_analyze(args, name):
+        tgt_dir = "systemd-analyze"
+        machine.succeed(
+            "systemd-analyze {} > {}/{} 2> {}/{}.err".format(
+                " ".join(args), tgt_dir, name, tgt_dir, name
+            )
```

Comment:
This seems to be overly generic. We only call it with one argument.

What about just executing `machine.succeed(…)` with a list of the "raw" commands:

```
machine.succeed(
    "systemd-analyze blame > blame.txt",
    "systemd-analyze critical-chain > critical-chain.txt",
    "systemd-analyze dot > dependencies.dot",
    "systemd-analyze plot > systemd-analyze.svg",
)

Reply:
> overly generic 

True, but cheap

> list of the "raw" commands

The replacement list loses stderr of the `dot` subcommand which does contain some data that could be in principle useful.

Also, I do want to have a subdirectory that I can copy as a whole.

Reply:
I don't care that much. I think it might be more readable if written more explicitly, but your call.

Reply:
Tried. It becomes pretty Black-fragile. Agree on the point of more comments needed.